### ISSUE TITLE
Runtime error

### ISSUE THREAD

**Issue by KBracknell** [skia4delphi/skia4delphi#200]
Hi, using Delphi 10.3 with Skia 4.1.

When I execute any SKIA-based EXE I get:
Parameter AContext cannot be nil.

Some solutions suggest a replacement unit for FMX.Graphics but this does not work.
Any ideas?

Thanks,
Kim.

**Comment by viniciusfbb** [comment]
Hi @KBracknell, thanks to report the issue!

Have you enabled the Skia in your project?
Inside the RAD Studio > right click on your project > “Enable Skia”

**Comment by KBracknell** [comment]
Thx viniciusfbb,
Good advice... the project had been set to have Skia enabled (as it initially showed as Skia disabled. However, Enabling Skia did not initially fix the problem. It did, however, work after a reboot (I had hadn't rebooted after initial Skia install). However, now if I open the and run the the Skia4Delphi demo that comes with the install it gives heaps of compile errors (yes, Skia is Enabled). Errors are as follows:

Messages
Checking project dependencies...
Compiling Skia4Delphi.dproj (Release. Win32)
brcc32 command line for "Skia4Delphi.vrc"
c:\program files (x86)\embarcadero\studio\20.0\bin\cgrc.exe -c65001 "Skia4Delphi.vrc" -foSkia4Delphi.res dcc32 command line for "Skia4Delphi.dpr"
(dcc32 Error] Sample.Form.Filter.pas(152): E2010 Incompatible types: 'Integer' and 'string'
[dcc32 Error] Sample.Form.Filter.pas(154): E2250There is no overloaded version of 'MakeShader' that can be called with these arguments [dcc32 Error] Sample.Form.Filter.pas(197): E2250 There is no overloaded version of 'Show' that can be called with these arguments [dcc32 Error] Sample.Form.Filter.pas(278): E2250 There is no overloaded version of 'Show' that can be called with these arguments [dcc32 Error] Sample.Form.Filter.pas(327): E2250 There is no overloaded version of 'Show' that can be called with these arguments [dcc32 Error] Sample.Form.Filter.pas(371): E2010 Incompatible types: 'Integer' and 'string'
[dcc32 Error] Sample.Form.Filter.pas(373): E2250 There is no overloaded version of 'MakeShader' that can be called with these arguments [dcc32 Error] Sample.Form.Filter.pas(416): E2250 There is no overloaded version of 'Show' that can be called with these arguments [dcc32 Error] Sample.Form.Filter.pas(464): E2250 There is no overloaded version of 'Show' that can be called with these arguments [dcc32 Error] Sample.Form.Filter.pas(491): E2010 Incompatible types: 'Integer' and 'string'
[dcc32 Error] Sample.Form.Filter.pas(493): E2250 There is no overloaded version of 'MakeShader' that can be called with these arguments [dcc32 Error] Sample.Form.Filter.pas(532): E2250 There is no overloaded version of 'Show' that can be called with these arguments [dcc32 Fatal Error] Skia4Delphi.dpr(17): F2063 Could not compile used unit 'Sample.Form.Filter.pas'
Failed
Elapsed time: 00:00:02.9
Build
Output
?
_________________________________________________________________________________________________
It almost looks like the 4.1 version of Skia is incompatibe with Delphi 10.3?

Any ideas?
Thanks again,
Kim.

**Comment by KBracknell** [comment]
I have since unistalled Skia 4.2 and reverted to 3.4 - demo mow compiles fine. I guess there are incompatibilities with more recent Skia versions? That's a shame...
Cheers,
Kim.

**Comment by viniciusfbb** [comment]
There is no incompatibility. Maybe it's a conflict with some old version installed. For a secure installation, follow these steps:

1. Install running the setup (the manual installation is not recommended) with the IDE closed. You can choose always the same destination folder (prefer the default folder to avoid administrator permission), even if there is an old installation there.
2. Delete old Skia4Delphi folders that you detect on your computer.
3. Check in your "RAD Studio > Options > Language > Delphi > Library > Library Path >" if there is any other skia directory besides these:
![image](https://user-images.githubusercontent.com/11139086/215127012-f0a5ddb2-f695-4647-bfe4-6b4fe1486f06.png)
4. Check in your "RAD Studio > Component > Install Package" if there is any other skia package besides these:
![image](https://user-images.githubusercontent.com/11139086/215127048-cf53705d-09c2-41d7-869f-6366fd1cb2db.png)

**Comment by viniciusfbb** [comment]
@KBracknell Recently a user had the same problem and we discovered that EtheaDev/SVGIconImageList internally adds a version of Skia4Delphi in its installation, resulting in 2 versions of the library in your RAD Studio. If you have this library, try to uninstall it, and reinstall Skia4Delphi.

I reported the issue to them few weeks ago and it's possible they already fixed it in their latest version.
